src/components/BackButton.js

```diff
@@ -10,127 +10,118 @@ import { Typography } from "@material-ui/core";
 import Grid from "@material-ui/core/Grid";
 
 
-class BackButton extends Component {
-	constructor(props) {
-		super(props);
-		this.state = {
-			alert: false,
-		};
-	}
+const BackButton = (props) => {
+
+	const [alert, setAlert] = React.useState(false);
 
-	label() {
+	const goBack = () => {
+		// only call user defined function from parent component if defined
+		props.onBack && props.onBack();
+		props.history.goBack();
+
+	}
+	const label = () => {
 		var label = "";
-		if (this.props.label == null) {
+		if (props.label == null) {
 			label = "Back";
 		}
-		if (this.props.label == "cancel") {
+		if (props.label == "cancel") {
 			label = "Cancel";
 		}
 		return label;
 	}
```

Comment:
See my earlier comments ... the button's label should be easily customizable, not hard-coded to a couple of presets